lib/src/database/sqflite_encryption_helper/io.dart

```diff
@@ -0,0 +1,218 @@
+    }
+    if (Platform.isMacOS) {
+      return DynamicLibrary.open(
+          '/usr/lib/libsqlcipher_flutter_libs_plugin.dylib');
+    }
+    if (Platform.isWindows) {
+      return DynamicLibrary.open('libsqlcipher.dll');
+    }
+
+    throw UnsupportedError('Unsupported platform: ${Platform.operatingSystem}');
+  }
+
+  /// checks whether the database exists and is encrypted
+  ///
+  /// In case it is not encrypted, the file is being migrated
+  /// to SQLCipher and encrypted using the given cipher and checks
+  /// whether that operation was successful
+  Future<void> ensureDatabaseFileEncrypted() async {
+    final file = File(path);
+
+    // in case the file does not exist there is no need to migrate
+    if (!await file.exists()) {
+      return;
+    }
+
+    // no work to do in case the DB is already encrypted
+    if (!await _isPlainText(file)) {
+      return;
+    }
+
+    Logs().d(
+        'Warning: Found unencrypted sqlite database. Encrypting using SQLCipher.');
+
+    // hell, it's unencrypted. This should not happen. Time to encrypt it.
+    final plainDb = await factory.openDatabase(path);
+
+    final encryptedPath = '$path.encrypted';
+
+    await plainDb.execute(
+        "ATTACH DATABASE '$encryptedPath' AS encrypted KEY '$cipher';");
+    await plainDb.execute("SELECT sqlcipher_export('encrypted');");
+    // ignore: prefer_single_quotes
+    await plainDb.execute("DETACH DATABASE encrypted;");
+    await plainDb.close();
+
+    Logs().d('Migrated data to temporary database. Checking integrity.');
+
+    final encryptedFile = File(encryptedPath);
+    // we should now have a second file - which is encrypted
+    assert(await encryptedFile.exists());
+    assert(!await _isPlainText(encryptedFile));
+
+    Logs().d('New file encrypted. Deleting plain text database.');
+
+    // deleting the plain file and replacing it with the new one
+    await file.delete();
+    await encryptedFile.copy(path);
+    // delete the temporary encrypted file
+    await encryptedFile.delete();
+
+    Logs().d('Migration done.');
+  }
+
+  /// safely applies the PRAGMA key to a [Database]
+  ///
+  /// To be directly used as [OpenDatabaseOptions.onConfigure].
+  ///
+  /// * ensures PRAGMA is supported by the given [database]
+  /// * applies [cipher] as PRAGMA key
+  /// * checks whether this operation was successful
+  Future<void> applyPragmaKey(Database database) async {
+    final cipherVersion = await database.rawQuery('PRAGMA cipher_version;');
+    if (cipherVersion.isEmpty) {
+      // Make sure that we're actually using SQLCipher, since the pragma
+      // used to encrypt databases just fails silently with regular
+      // sqlite3
+      // (meaning that we'd accidentally use plaintext databases).
+      throw StateError(
+        'SQLCipher library is not available, '
+        'please check your dependencies!',
+      );
+    } else {
+      Logs().d('PRAGMA supported by bundled SQLite. Encryption supported.');
```

Comment:
Just an idea, if we have a `cipherVersion` can we also print it here, like:

```dart
Logs().d('PRAGMA supported by bundled SQLite. Encryption supported with version: "$cipherVersion"');
```

Reply:
Sounds good, applied.